packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -215,6 +229,8 @@ const RelationModalRenderer = ({
 
   const currentDocumentMeta = state.documentHistory.at(-1) ?? rootDocumentMeta;
   const currentDocument = useDocument(currentDocumentMeta);
+  const isSingleType = currentDocumentMeta.collectionType === SINGLE_TYPES;
+  const isCreatingDocument = !currentDocumentMeta.documentId && !isSingleType;
```

Comment:
I see you need to define this in 3 places. I would recommend adding it to the relation context instead. Not in the reducer, but in the provider, next to all the other data that is derived from the reducer state